pkg/cloudprovider/aws/instancetypes.go

```diff
@@ -72,7 +72,15 @@ func (p *InstanceTypeProvider) Get(ctx context.Context, constraints *v1alpha1.Co
 	// Convert to cloudprovider.InstanceType
 	result := []cloudprovider.InstanceType{}
 	for _, instanceType := range instanceTypes {
-		instanceType.ZoneOptions = subnetZones.Intersection(instanceTypeZones[instanceType.Name()])
+		//TODO filter out possible zones and capacity types using an ICE cache
+		possibleZones := subnetZones.Intersection(instanceTypeZones[instanceType.Name()])
```

Comment:
any reason to not inline this?
```
for zone := range subnetZones.Intersection(instanceTypeZones[instanceType.Name()]) {
...
}
```

Reply:
No, this was an artifact of me re-working this code I think. I will fix it.